src/main/java/com/synopsys/integration/alert/common/descriptor/DescriptorMap.java

```diff
@@ -67,7 +67,7 @@ public DescriptorMap(final List<ChannelDescriptor> channelDescriptors, final Lis
         return descriptorMapping.values()
                    .stream()
                    .filter(descriptor -> descriptor.hasUIConfigForType(configType))
-                   .map(descriptor -> descriptor.getUIConfig(configType).generateDescriptorMetadata())
+                   .map(descriptor -> descriptor.getMetaData(configType).get())
```

Comment:
Since you return an optional here the get could cause a problem.  You probably need to something like this
.map(descriptor -> descriptor.getMetaData)
.filter(optional -> optional.isPresent)
.map(optional -> optional.get())

There may be another method for optional that you can use here as well.  But get() may throw an exception here.